### PR TITLE
Fixes amm

### PR DESCRIPTION
**Pull request by wwestgarth**
Three little niggles found while writing basic system tests for AMM.

### CODE REVIEW

File: core/collateral/engine.go [hunk 1]
```diff
@@ -3526,22 +3526,13 @@ func (e *Engine) CreatePartyAMMsSubAccounts(
...
 
-	_, err = e.CreatePartyGeneralAccount(ctx, subAccount, asset)
+	generalID, err := e.CreatePartyGeneralAccount(ctx, subAccount, asset)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	_, err = e.CreatePartyMarginAccount(ctx, subAccount, market, asset)
+	marginID, err := e.CreatePartyMarginAccount(ctx, subAccount, market, asset)
 	if err != nil {
```
Comment (jeremyletang) [inline core/collateral/engine.go:3536]: Any chance that would be doing something weird in the case a party had created an amm, cancelled it, creating another one again?

File: core/collateral/engine.go [hunk 2]
```diff
@@ -3526,22 +3526,13 @@ func (e *Engine) CreatePartyAMMsSubAccounts(
...
-		return nil, nil, errors.New("general sub account already exists")
-	}
-
-	marginID := e.accountID(market, subAccount, asset, types.AccountTypeMargin)
-	if _, ok := e.accs[marginID]; ok {
-		return nil, nil, errors.New("general sub account already exists")
-	}
 
-	_, err = e.CreatePartyGeneralAccount(ctx, subAccount, asset)
+	generalID, err := e.CreatePartyGeneralAccount(ctx, subAccount, asset)
```
Comment (EVODelavega) [inline core/collateral/engine.go:3530]: Is it not possible for a party to submit an AMM, cancel it, and submit a second one later on? In that case, it feels a bit off to call `CreatePartyGeneralAccount` and `CreatePartyMarginAccount`

**Review by EVODelavega — APPROVED** [review]

File: core/collateral/engine.go (hunk 1 above)
Comment (wwestgarth) [reply core/collateral/engine.go:3536]: I have a test that does exactly that and its all fine. Both `CreatePartyGeneralAccount()` and `CreatePartyMarginAccount()` just return the ID if the account already exists, doesn't try to re-create or anything

Comment (jeremyletang) [reply core/collateral/engine.go:3536]: sounds good then, my bad I didn't catch it in the test.

**Review by EVODelavega — APPROVED** [review]

**Pull request merged**